[WPE][GTK] Robustly handle subprocess leaks

carlosgcampos opened the pull request:
#### ee52175fab81d992a812d6b4ee1aee603a1412c8
<pre>
[WPE][GTK] Robustly handle subprocess leaks
<a href="https://bugs.webkit.org/show_bug.cgi?id=249272">https://bugs.webkit.org/show_bug.cgi?id=249272</a>

Reviewed by Michael Catanzaro.

Call _exit from a secondary thread after the connection is closed if the
main thread doesn&apos;t exit in 10 seconds.

* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::callExitSoon):
(WebKit::WebProcess::initializeConnection):

Canonical link: <a href="https://commits.webkit.org/258380@main">https://commits.webkit.org/258380@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7ddb178184c26e5d77d9dba927725748cac80ae6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101664 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10814 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34733 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110998 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11770 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1725 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94069 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108760 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107444 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8986 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92243 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/36561 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90879 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23665 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78555 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4412 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25161 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4487 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1611 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10575 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44649 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5739 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6241 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (commented):
I'd rather exit with a crash, e.g. using `std::abort()` rather than `_exit()`, because this indicates a bug and it's hard to know there's a bug if there's no core dump.

mcatanzaro reviewed (commented):
I wonder, is it possible to do this in AuxiliaryProcess instead, since it inherits from IPC::Connection::Client? Then all subprocess types would benefit, not just the web process?

carlosgcampos committed:
[WPE][GTK] Robustly handle subprocess leaks
https://bugs.webkit.org/show_bug.cgi?id=249272

Reviewed by Michael Catanzaro.

Call _exit from a secondary thread after the connection is closed if the
main thread doesn't exit in 10 seconds.

* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::callExitSoon):
(WebKit::WebProcess::initializeConnection):

Canonical link: https://commits.webkit.org/258380@main
